roxctl/common/connection.go

```diff
@@ -91,13 +93,35 @@ func addCommandHeaderStreamInterceptor(ctx context.Context, desc *grpc.StreamDes
 	return streamer(makeCtxWithCommandHeader(ctx), desc, cc, method, opts...)
 }
 
+func shouldRetry(err error) bool {
+	if errors.Is(err, context.DeadlineExceeded) || errors.Is(err, context.Canceled) {
+		return false // Do not retry on context errors.
+	}
+	if netErr, ok := err.(net.Error); ok && netErr.Timeout() {
+		return true // Retry on timeout errors.
+	}
+	if strings.Contains(err.Error(), "x509: certificate") {
+		return false // Do not retry on certificate errors.
+	}
```

Comment:
This may be leaky - any chance for using `error.Is` here?

Reply:
No, no chance for `error.Is`, unfortunately, due to the way how gRPC wraps the errors.